polygeist/include/mlir/Dialect/Polygeist/Analysis/UniformityAnalysis.h

```diff
@@ -66,17 +68,17 @@ class Uniformity {
   /// Create a state for \p val where the uniformity info is non-uniform.
   static Uniformity getNonUniform() { return Uniformity(Kind::NonUniform); }
 
-  /// Join two uniformity info
+  /// Join two uniformity info.
   static Uniformity join(const Uniformity &lhs, const Uniformity &rhs) {
     if (lhs.isUninitialized())
       return rhs;
     if (rhs.isUninitialized())
       return lhs;
     if (lhs == rhs)
       return lhs;
-    if (lhs.isNonUniform() && rhs.isUniform())
+    if (isNonUniform(lhs.getKind()) && isUniform(rhs.getKind()))
```

Comment:
I changed it because of the use in `anyModifierUniformityIs` which receive a kind but not a lattice. So it was easier to have a static function rather than a member function. 

I could also provide the equivalent member function...

Reply:
Added the member functions and simplified this part.